### PR TITLE
Improved docs for Loggers

### PR DESCRIPTION
**Pull request by awaelchli**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
A continuation of a series of doc improvements. Follows #1389, #1370, #1483 

- typos
- added links for classes, methods, modules etc. for easier navigation within the page.
- converted many code snippets to doctests and fixed failing ones.
- added instructions how to install the loggers #1487.

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/experiment_logging.rst [hunk 1]
```diff
@@ -1,201 +1,186 @@
 Experiment Logging
-===================
+==================
```
Comment (Borda) [inline docs/source/experiment_logging.rst:2]: not sure but I think that doctest in RST files are not executed, so there is no need of converting them here
https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html

File: pytorch_lightning/loggers/mlflow.py [hunk 2]
```diff
@@ -39,17 +19,38 @@ def any_lightning_module_function_or_hook(...):
 
 
 class MLFlowLogger(LightningLoggerBase):
+    """
+    Log using `MLflow <https://mlflow.org>`_
+
```
Comment (Borda) [inline pytorch_lightning/loggers/mlflow.py:24]:
```suggestion
    Example:
    
```

File: pytorch_lightning/loggers/comet.py [hunk 3]
```diff
@@ -33,57 +30,56 @@
 
 class CometLogger(LightningLoggerBase):
     r"""
-    Log using `comet.ml <https://www.comet.ml>`_.
+    Log using `Comet.ml <https://www.comet.ml>`_.
+    Requires either an API Key (online mode) or a local directory path (offline mode).
+
+    **ONLINE MODE**
```
Comment (Borda) [inline pytorch_lightning/loggers/comet.py:36]:
```suggestion
    Example:
    **ONLINE MODE**
```

File: pytorch_lightning/loggers/trains.py [hunk 4]
```diff
@@ -44,24 +23,45 @@ def any_lightning_module_function_or_hook(...):
 
 
 class TrainsLogger(LightningLoggerBase):
-    """Logs using TRAINS
+    """
+    Log using `allegro.ai TRAINS <https://github.com/allegroai/trains>`_.
+
```
Comment (Borda) [inline pytorch_lightning/loggers/trains.py:28]:
```suggestion

    Example:
```

**Comment by awaelchli** [comment]
@Borda 
Good point about doctest. I was wondering. 
I can run it locally like this for example: 
`python -m doctest docs/source/experiment_logging.rst`
Maybe we could add it to CI like this?
I will check the links you posted.

**Comment by Borda** [comment]
sure add it to CI :]

**Comment by awaelchli** [comment]
Hey @jakubczakon do you know how I could make the doctest pass in online mode without complaining about an invalid api key?

**Comment by jakubczakon** [comment]
Hi @awaelchli,

I think the reason could be:

- the project is not specified correctly. `api_token="ANONYMOUS"` works for some projects in the "shared" organization. We need to add that project by hand. That said "shared/pytorch-lightning-integration" is working.
- you are using an older version of neptune-client that doesn't yet have "ANONYMOUS" token hardcoded -> updating the project should fix it

Also, the fact that `api_token="test"` works in offline mode is because in offline mode nothing gets sent to Neptune so no token is checked etc.

Could you please paste the particular piece of code that is causing the problem?

**Comment by awaelchli** [comment]
@jakubczakon AWESOME. Your tipp with the version update did the trick! I didn't see it fail locally because I had the new version and the CI tests with minimal requirements. 
Thanks for your help!!

**Comment by awaelchli** [comment]
@Borda The doctest extension is already in `conf.py`. It can be run with `make doctest`. If I do locally, I get many import errors. Let's maybe fix it in another PR?

**Comment by Borda** [comment]
sounds fair to do it extra PR... I would maybe run in pytest then sphinx...

**Comment by Borda** [comment]
@awaelchli rebase to master shall fix it...

File: pytorch_lightning/loggers/mlflow.py [hunk 5]
```diff
@@ -40,21 +19,46 @@ def any_lightning_module_function_or_hook(...):
...
+    """
+    Log using `MLflow <https://mlflow.org>`_. Install it with pip:
+
+    .. code-block:: bash
 
+        pip install mlflow
+
+    Example:
+
+    >>> from pytorch_lightning import Trainer
```
Comment (Borda) [inline pytorch_lightning/loggers/mlflow.py:31]: this shall be shifted by 4 spaces

Comment (awaelchli) [reply pytorch_lightning/loggers/mlflow.py]: is it ok now?

**Comment by awaelchli** [comment]
@Borda about RST files: could I do that in another PR? on master "make doctest" has many errors (including RST) that would need to be solved and I think it would be better to do in another PR. What do you think?
https://github.com/PyTorchLightning/pytorch-lightning/pull/1484#issuecomment-613546668

**Comment by Borda** [comment]
so pls make it an issue linked to #714  so we do not forget... 
because now it seems as these examples shall be tested (because they are formatted as doctest) but in fact, they are not... lol

File: pytorch_lightning/loggers/comet.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/loggers/comet.py:43]: as example section, the doctest shall be shifted by 4 spaces...

File: pytorch_lightning/loggers/mlflow.py (hunk 5 above)
Comment (Borda) [reply pytorch_lightning/loggers/mlflow.py]: it is for this one but the same is in two more places
as Example is a section, the doctest shall be shifted, right?

**Comment by awaelchli** [comment]
I already added a note in #714 as a reminder 2 days ago. 
and the tests that I added here in rst I tested locally, with "make doctest docs/source/file.rst"

File: pytorch_lightning/loggers/mlflow.py (hunk 5 above)
Comment (awaelchli) [reply pytorch_lightning/loggers/mlflow.py]: ok, right. check again, is it better now?

**Pull request merged**